ui/apps/platform/src/Containers/NetworkGraph/simulation/NetworkPoliciesGenerationScope.tsx

```diff
@@ -26,54 +26,74 @@ function NetworkPoliciesGenerationScope({
     networkPolicyGenerationScope,
 }: NetworkPoliciesGenerationScopeProps) {
     const { granularity, cluster, deployments } = networkPolicyGenerationScope;
+    const [modalDeployments, setModalDeployments] = React.useState<
+        { namespace: string; name: string }[] | null
+    >(null);
 
     let deploymentElement = <span>All deployments</span>;
     let namespaceElement = <span>All namespaces</span>;
 
-    if (granularity === 'NAMESPACE' || granularity === 'DEPLOYMENT') {
+    if (granularity !== 'CLUSTER') {
         const namespaces = uniq(deployments.map((deployment) => deployment.namespace));
         const namespaceCount = namespaces.length;
         const namespaceText =
             namespaceCount === 1 ? `namespace "${namespaces[0]}"` : `${namespaceCount} namespaces`;
         namespaceElement = <span>{namespaceText}</span>;
-    }
 
-    if (granularity === 'DEPLOYMENT') {
         const deploymentCount = deployments.length;
         const deploymentText =
             deploymentCount === 1
                 ? `deployment "${deployments[0].name}"`
                 : `${deploymentCount} deployments`;
 
-        deploymentElement = <span>{deploymentText}</span>;
+        deploymentElement = (
+            <span>
+                <Button
+                    variant="link"
+                    isInline
+                    onClick={() => setModalDeployments(networkPolicyGenerationScope.deployments)}
+                >
+                    {deploymentText}
+                </Button>
+            </span>
+        );
```

Comment:
This simplifies the logic to only show "All namespaces"/"All deployments" when the only selected scope is the cluster.

Since requests for the graph data will not be sent unless a namespace is selected, we do not have detailed deployment data and cannot show the deployments in a modal.